server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -251,10 +251,27 @@ public SnapshotsService(
      * @param listener snapshot completion listener
      */
     public void executeSnapshot(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
-        createSnapshot(
-            request,
-            ActionListener.wrap(snapshot -> addListener(snapshot, ActionListener.map(listener, Tuple::v2)), listener::onFailure)
-        );
+        Repository repository = repositoriesService.repository(request.repository());
+        boolean remoteStoreIndexShallowCopy = REMOTE_STORE_INDEX_SHALLOW_COPY.get(repository.getMetadata().settings());
+        logger.debug("remote_store_index_shallow_copy setting is set as [{}]", remoteStoreIndexShallowCopy);
+
+        boolean isSnapshotV2 = SNAPSHOT_V2.get(repository.getMetadata().settings());
+        logger.debug("snapshot_v2 is set as [{}]", isSnapshotV2);
+
+        if (remoteStoreIndexShallowCopy
+            && clusterService.getClusterSettings().get(REMOTE_STORE_COMPATIBILITY_MODE_SETTING).equals(CompatibilityMode.MIXED)) {
+            // don't allow shallow snapshots if compatibility mode is not strict
+            logger.warn("Shallow snapshots are not supported during migration. Falling back to full snapshot.");
+            remoteStoreIndexShallowCopy = false;
+        }
+        if (remoteStoreIndexShallowCopy && isSnapshotV2 && request.indices().length == 0) {
```

Comment:
We have 2 options
- validation exception
- fallback to v1
I think it makes sense to fallback to v1 rather than failing the request because this is something caller doesn't care about

Reply:
Lets go with the fallback option.